App/Services/LeagueOfLegendsService.php

```diff
@@ -68,4 +70,39 @@ public function matchlist($summonerId)
         return $matchlist;
 
     }
+
+    /**
+     * @param $summonerId
+     * @return mixed
+     */
+    public function rawMatchlist($summonerId)
+    {
+        $summoner = new SummonersModel();
+        $matches = new MatchesModel();
```

Comment:
I generally keep models in the controllers, not in any other classes. What I would do with this method is convert their `matchlist()` class instance (on line 83) into a new class that we define. As we set properties on the new class we will mutate them beforehand.

Then this class would return our new class for matchlist and we would use that in the controller to persist to the database.